sensor: vl53l0x: add dts support ranging params

wuguanghai45 opened the pull request:
add dts option to set range profile

wuguanghai45 commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -381,7 +383,10 @@ static int vl53l0x_pm_action(const struct device *dev,
+		.xshut = GPIO_DT_SPEC_INST_GET_OR(inst, xshut_gpios, {}), \
+		.max_time_for_ranging = DT_INST_PROP_OR(inst, max_time_for_ranging, 33000),\
+		.signal_limit_ratio = DT_INST_PROP_OR(inst, signal_limit_ratio, 100), \
+		.sigma_limit_ratio = DT_INST_PROP_OR(inst, sigma_limit_ratio, 60000) \
```
changed

avisconti commented on dts/bindings/sensor/st,vl53l0x.yaml:
```diff
@@ -12,3 +26,25 @@ properties:
+    type: int
+    default: 33000
+    description: |
+      Set max time for ranging in μs
```
I think you need to explain the reason for default value.

wuguanghai45 replied:
ok

wuguanghai45 replied:
resolved

MaureenHelm replied:
The "explanation" just refers to a preprocessor macro defined in the driver. It doesn't actually explain why the value was selected. Is it a register reset value in the sensor?

Same comment applies to the additional new properties defined below.

avisconti reviewed (changes requested):
A small request but LGTM otherwise.
Can you also add in the PR description on what board you tested it?

wuguanghai45 committed:
sensor: vl53l0x: add dts support ranging params

add dts option to set range profile
support High accuracy, Long range, and High speed ranging or custom
Already tested on stm32l431rc

Signed-off-by: Guanghai Wu <wuguanghai45@126.com>

nashif commented:
@MaureenHelm  your change request addressed?

erwango commented:
@wuguanghai45 Please rebase
